barman/compression.py

```diff
@@ -574,13 +579,32 @@ def validate(self, pg_server_version, remote_status):
         issues = super(GZipPgBaseBackupCompressionOption, self).validate(
             pg_server_version, remote_status
         )
-        if self.config.level is not None and (
-            self.config.level < 1 or self.config.level > 9
+        levels = list(range(1, 10))
+        levels.append(-1)
+        if self.config.level is not None and remote_status[
+            "pg_basebackup_version"
+        ] < Version("15"):
+            # version prior to 15 allowed gzip compression 0
+            levels.append(0)
+            if self.config.level not in levels:
+                issues.append(
+                    "backup_compression_level %d unsupported by compression algorithm."
+                    " %s expects a compression level between -1 and 9 (-1 will use default compression level)."
+                    % (self.config.level, self.config.type)
+                )
+        if (
+            self.config.level is not None
+            and remote_status["pg_basebackup_version"] >= Version("15")
+            and self.config.level not in levels
         ):
-            issues.append(
-                "backup_compression_level %d unsupported by "
-                "pg_basebackup compression %s" % (self.config.level, self.config.type)
+            msg = (
+                "backup_compression_level %d unsupported by compression algorithm."
+                " %s expects a compression level between 1 and 9 (-1 will use default compression level)."
+                % (self.config.level, self.config.type)
             )
+            if self.config.level == 0:
+                msg += "\nIf you need to create an uncompress archive, you should set `backup_compression = none`."
```

Comment:
I would change "uncompress" with "not compressed"

Reply:
Agreed: will substitute `an uncompress archive` for `an archive not compressed`